lib/ansible/plugins/become/su.py

```diff
@@ -131,11 +131,15 @@ class BecomeModule(BecomeBase):
         '口令',
     ]
 
+    def __init__(self):
+        super(BecomeModule, self).__init__()
+        self._fields = {}
+
     def check_password_prompt(self, b_output):
         ''' checks if the expected password prompt exists in b_output '''
 
         prompts = self.get_option('prompt_l10n') or self.SU_PROMPT_LOCALIZATIONS
-        b_password_string = b"|".join((br'(\w+\'s )?' + to_bytes(p)) for p in prompts)
+        b_password_string = b"|".join((br'(\w+\'s )?' + to_bytes(p.format(self._fields))) for p in prompts)
```

Comment:
```suggestion
        b_password_string = b"|".join((br'(\w+\'s )?' + to_bytes(p.format(**self._fields))) for p in prompts)
```

`.format` requires splatting the dict, instead of just the dict itself